### DISCUSSION TITLE
Problem with linter after following Contribution docs

### DISCUSSION THREAD

**Discussion by Rishikesh01** [moby/moby#46969] [Q&A] [answered]
Hello folks, I have completed the dev setup after [following the docs/contributing](https://github.com/moby/moby/tree/master/docs/contributing), Yet still I face the problem with the linter not recognizing imports in vscode.

 ```
~/go/src/github.com/docker/docker> ls
api      client       CONTRIBUTING.md  Dockerfile.simple   hack             Jenkinsfile    MAINTAINERS  pkg       README.md       ROADMAP.md   vendor
AUTHORS  cmd          daemon           Dockerfile.windows  image            layer          Makefile     plugin    reference       runconfig    VENDORING.md
builder  codecov.yml  distribution     dockerversion       integration      libcontainerd  NOTICE       profiles  registry        SECURITY.md  vendor.mod
bundles  container    docker-bake.hcl  docs                integration-cli  libnetwork     oci          project   reports         TESTING.md   vendor.sum
cli      contrib      Dockerfile       errdefs             internal         LICENSE        opts         quota     restartmanager  testutil     volume
```

**Comment by thaJeztah** [comment]
Can you provide more details? Is this vscode's linter?

**Reply by Rishikesh01** [reply]
I am using vscode official Golang plugin which installs golps. What details exactly you need?


![Screenshot_2023-12-20_21-52-55](https://github.com/moby/moby/assets/53863619/58edd2df-4197-48a7-9269-23b1b017f7e2)

**Comment by thaJeztah** [comment, marked as answer]
Thanks! Hm, right, so I'm not a VSCode user myself, so don't know exactly how to configure it.

The general issue is that the repository is not yet a module (we use `vendor.mod` and the `go modules` _tools_, but we cannot yet add a `go.mod` for various reasons (being worked on, but really complicated!)

To make your IDE find the packages, you need to either configure it to run with go module support disabled (and have the correct `GOPATH` set up)

or (and this is a bit of a hack, but "it works") add `go.mod` and `go.sum` symlinks that point to the `vendor.mod` and `vendor.sum`. That's a bit of a hack (and some of our scripts _will_ print a warning), but "it works" as it makes your IDE see the code as a module (with vendoring). Be sure not to commit those files when contributing though!

To create the symlinks, you can use these commands (while in the repository's root directory);

```bash
ln -s vendor.mod go.mod
ln -s vendor.sum go.sum
```

**Reply by thaJeztah** [reply]
Whoops! Got the arguments for the symlinks reversed; I edited my comment above 🙈

**Reply by Rishikesh01** [reply]
Thank you. now it is working. Please consider adding it to [docs/contributing](https://github.com/moby/moby/tree/master/docs/contributing) it will help a lot of people I believe or I can make a PR if you want.

**Reply by thaJeztah** [reply]
Hehe, yes there was some back-and-forth on recommending the symlink trick, because "technically" it's not correct, and in some cases having a `go.mod` present can result in Go using a different strategy. (see https://github.com/moby/moby/pull/46942 and https://github.com/moby/moby/pull/46941 for some pretty "fun" things that Go 1.21 and up do)
